src/compiler/sys.ts

```diff
@@ -335,7 +335,12 @@ namespace ts {
                     data = "\uFEFF" + data;
                 }
 
-                _fs.writeFileSync(fileName, data, "utf8");
+                if (fileName === "/dev/stdout" || fileName === "-") {

```

Comment:
I'm not big on the special-casing here. This is also not necessarily cross-platform.